### ISSUE TITLE
.jar file creation.

### ISSUE THREAD

**Issue by sandeepcg** [nkh-lab/car-api-hello-world#2]
AOSP:
//      Add to Android.bp vehicle generated lib "vendor.nlab.vehicle-V1.0-java"

We saw above description in MainActivity.java to generate .jar file . Could you please provide the exact script ? 
I have tried below script ,but is not generating a jar file.
java_library {
    name: "vendor.nlab.vehicle-V1.0-java",
    srcs: [
        "types.hal",
    ],
}

**Comment by nkh-lab** [comment]
vendor.nlab.vehicle-V1.0-java.jar is Java generated code from your VHAL in your AOSP project. In the current example, it uses the VHAL of the NCAR AOSP project to work with custom vendor properties (VehiclePropertyView mVendorTest...). If you're only dealing with system properties that Android defines, you don't need them at all, or you can just use their magic numbers.
So it's not a bug, closed.

**Comment by sharmasadhna** [comment]
Hi @nkh-lab , it's not an issue but a development question from my side. Is it safe for the applications to use VHAL jar directly (include vendor.nlab.vehicle-V1.0-java.jar )? Is it not a security issue? In AOSP documentation they always say to use  CarPropertyManager to access any VHAL properties (i.e. modify */car-lib/src/android/car/VehiclePropertyIds.java to include new vendor properties and then use in the app code). Can you please elaborate on this?

**Comment by nkh-lab** [comment]
Hi @sharmasadhna, as far as I remember this jar contains only one file VehicleProperty.java which defines only properties (constants) (e.g. `public static final int INFO_FUEL_DOOR_LOCATION = 289407240`). So this is just for user convenience, if you prefer to work with raw magic numbers you can use them directly :)

**Comment by sharmasadhna** [comment]
@nkh-lab , thank you for your reply. Let me rephrase the question. Is it a better way if we copy the properties constants from generated  vendor.nlab.vehicle-V1.0-java.jar (VehicleProperty.java) to VehiclePropertyIds.java in car-lib (Car/car-lib/src/android/car/VehiclePropertyIds.java). and then use modified car-lib for applications?. My doubt is coming from the fact that  vendor.nlab.vehicle-V1.0-java.jar, is a part of HAL and AOSP documentation says we should not access HAL directly in the apk but via car-lib/CarService.
Sorry for the long post. But I am stuck, and will apppreciate your help here.

**Comment by nkh-lab** [comment]
In my opinion I would use this library to avoid manual copy and paste, I don't see any violations, especially if you a part of whole AOSP build.

**Comment by sharmasadhna** [comment]
@nkh-lab , so you mean if application is also part of AOSP, its good to use this jar? But what about the third paty application developers, it's ok to share the jar with them as well right?

**Comment by nkh-lab** [comment]
I don't see any problem, since this jar and especially VehicleProperty.java only does the job of translating the magic numbers of the properties into a human-readable format. Either you can provide this information through this library, or provide this information in another format (some supporting document).

**Comment by sharmasadhna** [comment]
@nkh-lab , oh great, Thanks for the discussion.
